matchbox_signaling/examples/server.rs

```diff
@@ -0,0 +1,27 @@
+use matchbox_signaling::SignalingServer;
+use std::net::Ipv4Addr;
+use tracing::info;
+
+#[tokio::main]
+async fn main() -> Result<(), matchbox_signaling::Error> {
+    // Setup logging
+    use tracing_subscriber::prelude::*;
+    tracing_subscriber::registry()
+        .with(
+            tracing_subscriber::EnvFilter::try_from_default_env()
+                .unwrap_or_else(|_| "debug".into()),
+        )
+        .with(tracing_subscriber::fmt::layer())
+        .init();
+
+    let server = SignalingServer::client_server_builder((Ipv4Addr::UNSPECIFIED, 3536))
+        .on_host_connected(|id| info!("Host joined: {id:?}"))
+        .on_host_disconnected(|id| info!("Host left: {id:?}"))
+        .on_client_connected(|id| info!("Client joined: {id:?}"))
+        .on_client_disconnected(|id| info!("Client left: {id:?}"))
+        .on_signal(|s| info!("Signal: {s:?}"))
+        .cors()
+        .trace()
+        .build();
+    server.serve().await
+}
```

Comment:
As above, it would be nice to move logging setup out of main for the sake of clarity

```suggestion
#[tokio::main]
async fn main() -> Result<(), matchbox_signaling::Error> {
    setup_logging();

    let server = SignalingServer::full_mesh_builder((Ipv4Addr::UNSPECIFIED, 3536))
        .on_peer_connected(|id| info!("Joined: {id:?}"))
        .on_peer_disconnected(|id| info!("Left: {id:?}"))
        .on_signal(|s| info!("Signal: {s:?}"))
        .cors()
        .trace()
        .build();
    server.serve().await
}

fn setup_logging() {
    use tracing_subscriber::prelude::*;
    tracing_subscriber::registry()
        .with(
            tracing_subscriber::EnvFilter::try_from_default_env()
                .unwrap_or_else(|_| "debug".into()),
        )
        .with(tracing_subscriber::fmt::layer())
        .init();
}
```